octodns_googlecloud/__init__.py

```diff
@@ -203,6 +207,27 @@ def _get_gcloud_records(self, gcloud_zone, page_token=None):
                 # yield from is in python 3 only.
                 yield gcloud_record
 
+    def _is_zone_private(self, zone: ManagedZone) -> bool:
+        """Determine if a ManagedZone is private.
+        :param zone: zone to check
+        :type zone: google.cloud.dns.ManagedZone
+
+        :return: True if the zone is private, False otherwise
+        """
+        return zone._properties.get('visibility', '') == 'private'
+
+    def _filter_zone(self, zone: ManagedZone) -> bool:
+        """Determine if the zone matches filtering criteria.
+
+        :param zone: zone to check
+        :type zone: google.cloud.dns.ManagedZone
+
+        :return: True if the zone should be included, False otherwise
+        """
+        if self.private is None:
+            return True
+        return self.private == self._is_zone_private(zone)
```

Comment:
I think this probably reads cleaner as a one-liner. 

```suggestion
        return self.private is None or self.private == self._is_zone_private(zone)
```